Boltdb cannot query documents that have ids with underscores

I used ids with underscores like: "prod_LK8gsiRhdFpzJB" from stripe.

It sets ok but when performing queries the following error occurs due to the splitting of underscores.

```

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0xdd995c]

goroutine 199 [running]:
github.com/nitrictech/nitric/pkg/plugins/document/boltdb.toSdkDoc(0xc0006d54d0, {{0xc0002b3f68, 0x13}, {0xc0002b3f80, 0x13}, {0xc000808670, 0x9}, 0xc000bffec0})
        C:/Users/davem/go/pkg/mod/github.com/nitrictech/nitric@v0.15.0-rc.9/pkg/plugins/document/boltdb/boltdb.go:484 +0xfc
github.com/nitrictech/nitric/pkg/plugins/document/boltdb.(*BoltDocService).query(0x204b6c0, 0xc0006d54d0, {0x4d2b5f0, 0x0, 0x0}, 0x0, 0x0, 0xc00015fa98)
        C:/Users/davem/go/pkg/mod/github.com/nitrictech/nitric@v0.15.0-rc.9/pkg/plugins/document/boltdb/boltdb.go:333 +0x104e
github.com/nitrictech/nitric/pkg/plugins/document/boltdb.(*BoltDocService).Query(0x0, 0xc0006d54d0, 
{0x4d2b5f0, 0x0, 0x0}, 0x27758ddbf40, 0x279d6d)
        C:/Users/davem/go/pkg/mod/github.com/nitrictech/nitric@v0.15.0-rc.9/pkg/plugins/document/boltdb/boltdb.go:363 +0x114
github.com/nitrictech/nitric/pkg/adapters/grpc.(*DocumentServiceServer).Query(0xc0006d03a0, {0xc0000cfaa0, 0x39f506}, 0xc0000cfaa0)
        C:/Users/davem/go/pkg/mod/github.com/nitrictech/nitric@v0.15.0-rc.9/pkg/adapters/grpc/document_grpc.go:123 +0x103
github.com/nitrictech/nitric/pkg/api/nitric/v1._DocumentService_Query_Handler({0x22631e0, 0xc0006d03a0}, {0x2b31068, 0xc000bffbf0}, 0xc0000cfa40, 0x0)
        C:/Users/davem/go/pkg/mod/github.com/nitrictech/nitric@v0.15.0-rc.9/pkg/api/nitric/v1/document_grpc.pb.go:222 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000793340, {0x2b819e0, 0xc000604820}, 0xc000eed440, 0xc000ac4360, 0x4cb0308, 0x0)
        C:/Users/davem/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000793340, {0x2b819e0, 0xc000604820}, 0xc000eed440, 0x0)
        C:/Users/davem/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1616 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        C:/Users/davem/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        C:/Users/davem/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:919 +0x294
```

### System
Go Version: go1.17.5
Go OS/Arch: windows/amd64
Git commit: 0c87eb6
Build time: 2022-03-14T10:02:24+1100
Nitric CLI: v1.1.0-develop.14-73-g0c87eb6f6866723aa1a2831049959b1fb0cc3c6f

--- a/pkg/plugins/document/boltdb/boltdb.go
+++ b/pkg/plugins/document/boltdb/boltdb.go
@@ -35,11 +35,12 @@ import (
 )
 
 const DEV_SUB_DIRECTORY = "./collections/"
-
-const skipTokenName = "skip"
-const idName = "Id"
-const partitionKeyName = "PartitionKey"
-const sortKeyName = "SortKey"
+const (
+	skipTokenName    = "skip"
+	idName           = "Id"
+	partitionKeyName = "PartitionKey"
+	sortKeyName      = "SortKey"
+)
 
 type BoltDocService struct {
 	document.UnimplementedDocumentPlugin
@@ -465,7 +466,7 @@ func createDoc(key *document.Key) BoltDoc {
 		}
 	} else {
 		return BoltDoc{
-			Id:           parentKey.Id + "_" + key.Id,
+			Id:           parentKey.Id + document.SubcollectionDelimiter + key.Id,
 			PartitionKey: parentKey.Id,
 			SortKey:      key.Collection.Name + "#" + key.Id,
 		}
@@ -473,7 +474,7 @@ func createDoc(key *document.Key) BoltDoc {
 }
 
 func toSdkDoc(col *document.Collection, doc BoltDoc) *document.Document {
-	keys := strings.Split(doc.Id, "_")
+	keys := strings.Split(doc.Id, document.SubcollectionDelimiter)
 
 	// Translate the boltdb Id into a nitric document key Id
 	var id string

--- a/pkg/plugins/document/document.go
+++ b/pkg/plugins/document/document.go
@@ -20,6 +20,8 @@ import (
 	"strings"
 )
 
+const SubcollectionDelimiter = "+"
+
 // Map of valid expression operators
 var validOperators = map[string]bool{
 	"==":         true,
@@ -37,6 +39,9 @@ func ValidateKey(key *Key) error {
 	}
 	if key.Id == "" {
 		return fmt.Errorf("provide non-blank key.Id")
+	}
+	if strings.Contains(key.Id, SubcollectionDelimiter) {
+		return fmt.Errorf("key.Id cannot contain %s", SubcollectionDelimiter)
 	}
 	if key.Collection == nil {
 		return fmt.Errorf("provide non-nil key.Collection")

